OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -314,25 +322,43 @@ bool LoadAsset(string filename)
 			if (!modData.DefaultFileSystem.Exists(filename))
 				return false;
 
-			if (Path.GetExtension(filename.ToLowerInvariant()) == ".vqa")
-			{
-				player = panel.Get<VqaPlayerWidget>("PLAYER");
-				currentFilename = filename;
-				player.Load(filename);
-				player.DrawOverlay = false;
-				isVideoLoaded = true;
-				frameSlider.MaximumValue = (float)player.Video.Frames - 1;
-				frameSlider.Ticks = 0;
-				return true;
-			}
-			else
+			errorLabelWidget.Visible = false;
+
+			try
 			{
+				if (Path.GetExtension(filename.ToLowerInvariant()) == ".vqa")
+				{
+					player = panel.Get<VqaPlayerWidget>("PLAYER");
+					currentFilename = filename;
+					player.Load(filename);
+					player.DrawOverlay = false;
+					isVideoLoaded = true;
+					frameSlider.MaximumValue = (float)player.Video.Frames - 1;
+					frameSlider.Ticks = 0;
+					return true;
+				}
+
 				currentFilename = filename;
 				currentSprites = world.Map.Rules.Sequences.SpriteCache[filename];
 				currentFrame = 0;
 				frameSlider.MaximumValue = (float)currentSprites.Length - 1;
 				frameSlider.Ticks = currentSprites.Length;
 			}
+			catch (Exception ex)
+			{
+				if (errorLabelWidget != null)
+				{
+					errorLabelWidget.Text = WidgetUtils.TruncateText(errorLabelWidget.Text.Replace("{{filename}}", filename),
+						errorLabelWidget.Bounds.Width, errorFont);
+
+					currentSprites = new Sprite[0];
+					errorLabelWidget.Visible = true;
+				}
+
+				Log.Write("debug", "Error reading {0}: {1} {2}", filename, ex.Message, ex.StackTrace);
```

Comment:
debug.log is pretty crowded even in the main menu due to the shellmap and we don't really care at which game tick this happened. Maybe use exception.log or invent your own new channel.